Add hook to conditionally show the lesson actions block in the frontend

Imran92 opened the pull request:
Resolves https://github.com/Automattic/sensei-pro/issues/1956 (partial)

## Proposed Changes
We've added a hook here which will allow us to conditionally render the lesson actions block in the frontend from elsewhere (i.e. Sensei-pro)

## Testing Instructions
<!--
Add as many details as possible to help others reproduce the issue and test the changes.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

1. Follow the instructions in - https://github.com/Automattic/sensei-pro/pull/2607

## New/Updated Hooks
<!-- Add the following only if there are new/updated actions or filters. Please provide a brief description of what they do and any arguments they may take. Be sure to also add the "Hooks" label to this PR. -->

- **sensei_lesson_show_actions** - Flag to show or hide the lesson actions block in the frontend.

## Pre-Merge Checklist
<!-- Complete applicable items on this checklist **before** merging. Items that are not applicable can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed. -->
- [x] PR title and description contain sufficient detail and accurately describe the changes
- [ ] Acceptance criteria is met
- [x] Decisions are publicly documented
- [x] Adheres to coding standards ([PHP](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/php/), [JavaScript](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/), [CSS](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/css/), [HTML](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/html/))
- [ ] All strings are translatable (without concatenation, handles plurals)
- [x] Follows our naming conventions (P6rkRX-4oA-p2)
- [x] Hooks (p6rkRX-1uS-p2) and functions are documented
- [ ] New UIs are responsive and use a [mobile-first approach](https://zellwk.com/blog/how-to-write-mobile-first-css/)
- [ ] New UIs match the designs
- [x] Different user privileges (admin, teacher, subscriber) are tested as appropriate
- [ ] Legacy courses (course without blocks) are tested
- [x] Code is tested on the minimum supported PHP and WordPress versions
- [x] User interface changes have been tested on the latest versions of Chrome, Firefox and Safari
- [ ] "Needs Documentation" label is added if this change requires updates to documentation
- [ ] Known issues are created as new GitHub issues

Imran92 committed:
Add hook to prevent rendering lesson actions

Imran92 committed:
Add tests to make sure the hook is working

Imran92 committed:
Add changelog

donnapep commented on tests/unit-tests/test-class-lesson.php:
```diff
@@ -1358,6 +1358,21 @@ public function testShouldShowLessonActions_WhenPrerequisiteFoundAndWasNotLesson
 		self::assertFalse( $result );
 	}
 
+	public function testShouldShowLessonActions_WhenHookSetsFalce_ReturnsFalse(): void {
```
This test case name is somewhat misleading. In particular, the `WhenHookSetsFalse` part because in this test case we're testing both `true` and `false`. We could split these into 2 separate use cases to test each since there isn't much repetitive setup code.

donnapep reviewed (dismissed):
Looks good. I added a small comment about the tests though.

Imran92 committed:
Split a test in two

Imran92 committed:
Merge branch 'trunk' into fix/prevent-rendering-lesson-actions-in-content-drip

Imran92 committed:
Fix test

Imran92 committed:
Test the version

Imran92 committed:
Remove test lines

Imran92 committed:
Merge branch 'trunk' into fix/prevent-rendering-lesson-actions-in-content-drip

Imran92 commented:
> Looks good. I added a small comment about the tests though.

Thanks @donnapep ! I've updated it here https://github.com/Automattic/sensei/pull/7677/commits/438d8cabf4f5f2db521703dcccc7c4e6a010517b . Can you kindly take a look again? Thanks!

donnapep reviewed (approved):
🚀
